internal/core/application/walletunlocker_service.go

```diff
@@ -609,7 +602,28 @@ func (w *walletUnlockerService) restoreMarket(
 			}
 		}
 
-		if err := market.FundMarket(outpoints, w.marketBaseAsset); err != nil {
+		outpointsByAsset := make(map[string]struct{})
+		for _, o := range outpoints {
+			outpointsByAsset[o.Asset] = struct{}{}
+		}
+		var baseAsset, quoteAsset string
+		for asset := range outpointsByAsset {
+			if asset == w.marketBaseAsset {
+				baseAsset = asset
+			} else {
```

Comment:
what will happens if my mistake the user deposits 3 assets types? we should eventually check explicitly for quoteAsset here

Reply:
Here we're in the process of restoring a daemon wallet and because depositing 3 asset types in a market is an operation not allowed by the daemon, we suppose here that all the markets were previously created in the right way.

If we don't want to be so optimistic, here the only thing we can do is returning an error because the daemon cannot be sure of the asset to use the quote one of the market.

Reply:
> we should eventually check explicitly for quoteAsset here

This is not possible at this stage. In this function, we're restoring the markets, which means that we don't know the quote assets of each market found and we need to discover them. 
`w.marketBaseAsset` is basically the configurable base asset from env var. It's really helpful in this process because it allows the daemon to figure out the market pairs of every market account to be restored.